aws/resource_aws_ecr_registry_policy.go

```diff
@@ -0,0 +1,117 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAwsEcrRegistryPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEcrRegistryPolicyPut,
+		Read:   resourceAwsEcrRegistryPolicyRead,
+		Update: resourceAwsEcrRegistryPolicyPut,
+		Delete: resourceAwsEcrRegistryPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"policy": {
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: suppressEquivalentAwsPolicyDiffs,
+				ValidateFunc:     validation.StringIsJSON,
+			},
+			"registry_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsEcrRegistryPolicyPut(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ecrconn
+
+	input := ecr.PutRegistryPolicyInput{
+		PolicyText: aws.String(d.Get("policy").(string)),
+	}
+
+	log.Printf("[DEBUG] Creating ECR resository policy: %s", input)
+
+	// Retry due to IAM eventual consistency
+	var err error
+	var out *ecr.PutRegistryPolicyOutput
+	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
```

Comment:
If this is an IAM eventual consistency wait, use `iamwaiter.PropagationTimeout`:
https://github.com/hashicorp/terraform-provider-aws/issues/16752